lib/librte_gro/rte_gro.c

```diff
@@ -242,17 +304,26 @@ rte_gro_timeout_flush(void *ctx,
 {
 	struct gro_ctx *gro_ctx = ctx;
 	uint64_t flush_timestamp;
+	uint16_t num = 0;
 
 	gro_types = gro_types & gro_ctx->gro_types;
 	flush_timestamp = rte_rdtsc() - timeout_cycles;
 
-	if (gro_types & RTE_GRO_TCP_IPV4) {
-		return gro_tcp4_tbl_timeout_flush(
+	if (gro_types & RTE_GRO_IPV4_VXLAN_TCP_IPV4) {
+		num = gro_vxlan_tcp4_tbl_timeout_flush(gro_ctx->tbls[
+				RTE_GRO_IPV4_VXLAN_TCP_IPV4_INDEX],
+				flush_timestamp, out, max_nb_out);
+	}
+
+	max_nb_out -= num;
```

Comment:
What is this line for? Can gro_types set with RTE_GRO_IPV4_VXLAN_TPC_IPV4 and RTE_GRO_TCP_IPV4 both?

Reply:
The logical also seems a bit confused in this function, I suggest to break into two small function to do one thing well.